src/support/virtualDom.ts

```diff
@@ -0,0 +1,109 @@
+import { assign } from '@dojo/core/lang';
+import { DNode, HNode, VirtualDomProperties, WidgetProperties, WNode } from '@dojo/widget-core/interfaces';
+import { isHNode, isWNode } from '@dojo/widget-core/d';
+
+export function assignChildProperties(target: WNode | HNode, index: number | string, properties: WidgetProperties | VirtualDomProperties): WNode | HNode {
+	const node = findIndex(target, index);
+	if (!node || !(isWNode(node) || isHNode(node))) {
+		throw new TypeError(`Index of "${index}" is not resolving to a valid target`);
+	}
+	assignProperties(node, properties);
+	return target;
+}
+
+export function assignProperties(target: HNode, properties: VirtualDomProperties): HNode;
+export function assignProperties(target: WNode, properties: WidgetProperties): WNode;
+export function assignProperties(target: WNode | HNode, properties: WidgetProperties | VirtualDomProperties): WNode | HNode;
+export function assignProperties(target: WNode | HNode, properties: WidgetProperties | VirtualDomProperties): WNode | HNode {
+	assign(target.properties, properties);
+	return target;
+}
+
+/**
+ * Replace a child of DNode.
+ *
+ * *NOTE:* The replacement modify the passed `target` and does not return a new instance of the `DNode`.
+ * @param target The DNode to replace a child element on
```

Comment:
http://grammar.yourdictionary.com/parts-of-speech/prepositions/Ending-a-Sentence-with-a-Preposition.html